components/thanks_io/sources/common/common.mjs

```diff
@@ -0,0 +1,54 @@
+import thanksIo from "../../thanks_io.app.mjs";
+
+export default {
+  props: {
+    thanksIo,
+    db: "$.service.db",
+    timer: {
+      label: "Polling interval",
+      description: "Pipedream will poll the API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    subAccount: {
+      propDefinition: [
+        thanksIo,
+        "subAccount",
+      ],
+    },
+  },
+  methods: {
+    _getLastTimestamp() {
+      return this.db.get("lastTimestamp") || null;
+    },
+    _setLastTimestamp(lastTimestamp) {
+      this.db.set("lastTimestamp", lastTimestamp);
+    },
+    isLater(newDate, compareDate) {
+      return compareDate
+        ? (newDate > compareDate)
+        : true;
+    },
+    async paginate(resourceFn, params) {
```

Comment:
Don't you think this function should handle max number of pages? in case the result ends up having too many pages?